contracts/alliance-nft-collection/src/contract/execute.rs

```diff
@@ -345,29 +437,46 @@ fn try_change_owner(
     ]))
 }
 
+fn try_update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    msg: UpdateConfigMsg,
+) -> Result<Response, ContractError> {
+    let mut cfg = CONFIG.load(deps.storage)?;
+    authorize_execution(cfg.owner.clone(), info.sender)?;
+
+    if let Some(dao_treasury_address) = msg.dao_treasury_address {
+        cfg.dao_treasury_address = deps.api.addr_validate(&dao_treasury_address)?;
+    }
+
+    if let Some(dao_treasury_share) = msg.dao_treasury_share {
+        cfg.dao_treasury_share = validate_dao_treasury_share(dao_treasury_share)?;
+    }
+
+    if let Some(set_whitelisted_reward_assets) = msg.set_whitelisted_reward_assets {
+        let mut set_assets =
+            validate_whitelisted_assets(&deps, &cfg.lst_asset_info, set_whitelisted_reward_assets)?;
+        dedupe_assetinfos(&mut set_assets);
+        cfg.whitelisted_reward_assets = set_assets;
+    }
+
+    if let Some(add_whitelisted_reward_assets) = msg.add_whitelisted_reward_assets {
+        let mut add_assets =
+            validate_whitelisted_assets(&deps, &cfg.lst_asset_info, add_whitelisted_reward_assets)?;
+        let mut existing = cfg.whitelisted_reward_assets;
+        existing.append(&mut add_assets);
+        dedupe_assetinfos(&mut existing);
+        cfg.whitelisted_reward_assets = existing;
+    }
```

Comment:
Seems like here you can only whitelist, but is there any plan to also remove tokens from the whitelist? As larger the whitelist more gas will be consumed when users execute actions

Reply:
It was planned to just use the "set_whitelisted_reward_assets" if some tokens should be removed. Theoretically the "add_whitelisted_reward_assets" could also be removed to reduce contract complexity, but it is always simpler to use an "add delta", which was always an issue with astroport incentive proposals, so making it simpler to add tokens, but if something is going to be removed, always enforce sending the whole new set of assets.